geonode/base/forms.py

```diff
@@ -554,6 +555,38 @@ class BatchPermissionsForm(forms.Form):
     )
 
 
+class UserAndGroupPermissionsForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        super(UserAndGroupPermissionsForm, self).__init__(*args, **kwargs)
+        self.fields['layers'].label_from_instance = self.label_from_instance
```

Comment:
When is this called?

Reply:
@petrus7, The above line is part of the class constructor, it is called within the constructor and overides the '__str__' method of the 'Layer' model field for`UserAndGroupPermissionsForm` only.